src/test/java/teammates/test/util/TestUrl.java

```diff
@@ -0,0 +1,16 @@
+package teammates.test.util;
+
+import teammates.common.util.AppUrl;
+import teammates.test.driver.TestProperties;
+
+/**
+ * An extension of {@link AppUrl}, differing in the base URL used for conversion.
+ * In this class, the base URL used is the value of test.app.url in test.properties.
+ */
+public class TestUrl extends AppUrl {
```

Comment:
TestAppUrl? It's consistent with the hierarchy. Url <-- AppUrl <-- TestAppUrl 
AppUrl and TestUrl sounds like they are at the same level.